doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -471,6 +471,39 @@ Field              | Type       | Meaning
 `cpu_nano_cores`   | [`bigint`] | Approximate CPU usage, in billionths of a vCPU core.
 `memory_bytes`     | [`bigint`] | Approximate RAM usage, in bytes.
 
+### `mz_storage_source_statistics`
+
+The `mz_storage_source_statistics` table gives the last known storage-specific statistics
+for all processes of all extant storage sources.
+
+At this time, we do not make any guarantees about the exactness or freshness of these numbers.
+
+Field                 | Type        | Meaning
+----------------------|-------------|--------
+`id`                 | [`text`]     | The ID of the storage object (source or sink).
+`worker_id`          | [`bigint`]   | An identifier of a process within a host.
+`snapshot_committed` | [`boolean`]  | Whether or not we have committed the initial snapshot for a source.
+`messages_received`  | [`bigint`]   | How many messages we have received from upstream.
+`messages_committed` | [`bigint`]   | How many messages we have committed into our storage layer.
+`bytes_received`     | [`bigint`]   | How many bytes worth of messages we have received from upstream.
+
+These values all require some amount of interpretation:
+
+- `messages_received` does not distinguish between inserts and deletes. Some upstream sources, like postgres,
+  produce deletion messages that we count the same as insertions.
+- `messages_committed` can be permanently lower or higher than `messages_received`, because:
+  - The storage layer can choose to consolidate messages within a certain time interval.
+  - Some envelopes (like `UPSERT`) can produce 2 messages (an insert and a delete) from a single upstream message.
```

Comment:
I think we need to go back to the drawing board on this. IMO we can't have `messages_received` and `messages_committed` count messages differently, as they both use "messages" in their name. The metric as described is, I think, better named something like `updates_committed`... but I don't think that's necessarily useful to expose? (In the future, I think `updates_committed` is computable with `SELECT count(*) FROM changes(src)`, and in the meantime I think `SELECT count(*) FROM src` is a decent proxy for `updates_committed`.)

Put another way: the value in exposing these two quantities seemed to be that you could watch for skew between `messages_received and `messages_committed` to detect problems in the source ingestion pipeline. If these two metrics are expected to skew, they seem substantially less useful, and at the very least seems like we need to rename them so that they do not seem so comparable.

Reply:
strawman: we could have three statistics:
 - `messages_received`: the raw messages that we get from the source
 - `updates` (name tbd) number of updates that we have _after_ messages have gone through the enveloping logic
 - `updates_committed`
 
Depending on the envelope, there will be systemic skew between `messages_received` and `updates`, but with a constant factor/bound. On the other hand,  `updates` and `updates_committed` should proceed in lockstep and you will notice that something is afoot when they don't. wdyt?

Reply:
cc my usual crew for these user-facing things @morsapaes, @sjwiesman, @joacoc, @ggnall  👋

Reply:
cc my usual crew for these user-facing things @morsapaes, @sjwiesman, @joacoc, @ggnall  👋

Reply:
I think @aljoscha 's take makes sense. Idk what the right names are, I'll need to think about it.

Reply:
@aljoscha `updates` can be incremented when we write the updates into (non-committed) persist batches, so it will always move in lock-step with `updates_committed`

Sounds like a reasonable plan to me!

Reply:
Updated the pr to reflect this (with the given placeholder names)

Reply:
I like it! Maybe `updates_staged` instead of `updates`?

Reply:
+1 for `updates_staged`.